tests/transformers/test_transformers_prompt_templating.py

```diff
@@ -0,0 +1,171 @@
+from unittest.mock import MagicMock
+
+import pytest
+import yaml
+from transformers import pipeline
+
+import mlflow
+from mlflow.exceptions import MlflowException
+from mlflow.models.model import MLMODEL_FILE_NAME
+from mlflow.transformers import _SUPPORTED_PROMPT_TEMPLATING_TASK_TYPES
+from mlflow.utils.model_utils import _validate_prompt_template
+
+# session fixtures to prevent saving and loading a ~400mb model every time
+TEST_PROMPT_TEMPLATE = "Answer the following question like a pirate:\nQ: {prompt}\nA: "
+
+UNSUPPORTED_PIPELINES = [
+    "audio-classification",
+    "automatic-speech-recognition",
+    "text-to-audio",
+    "text-to-speech",
+    "text-classification",
+    "sentiment-analysis",
+    "token-classification",
+    "ner",
+    "question-answering",
+    "table-question-answering",
+    "visual-question-answering",
+    "vqa",
+    "document-question-answering",
+    "translation",
+    "zero-shot-classification",
+    "zero-shot-image-classification",
+    "zero-shot-audio-classification",
+    "conversational",
+    "image-classification",
+    "image-segmentation",
+    "image-to-text",
+    "object-detection",
+    "zero-shot-object-detection",
+    "depth-estimation",
+    "video-classification",
+    "mask-generation",
+    "image-to-image",
+]
+
+
```

Comment:
We probably want to include the fixtures `clean_cache` and `force_gc` from the main transformers test suite as they handle unrecoverable disk and heap cleanup on the linux vm runners. The suites were super flaky before we added these fixtures.